src/main/kotlin/com/google/androidstudiopoet/generators/DependencyGraphGenerator.kt

```diff
@@ -0,0 +1,166 @@
+package com.google.androidstudiopoet.generators
+
+import com.google.androidstudiopoet.models.*
+import com.google.androidstudiopoet.utils.joinPath
+import com.google.androidstudiopoet.writers.FileWriter
+import com.google.androidstudiopoet.writers.ImageWriter
+import java.awt.Color
+import java.awt.Graphics
+import java.awt.image.BufferedImage
+
+class DependencyGraphGenerator(private val fileWriter: FileWriter, private val imageWriter: ImageWriter) {
+
+    fun generate(projectBlueprint: ProjectBlueprint) {
+        val dependenciesGraphFileName = projectBlueprint.projectRoot.joinPath("dependencies.dot")
+        fileWriter.writeToFile(dependenciesGraphString(projectBlueprint), dependenciesGraphFileName)
+        println("Dependency graph written to $dependenciesGraphFileName")
+        DependencyImageGenerator(imageWriter).generate(projectBlueprint)
+    }
+
+    private fun dependenciesGraphString(projectBlueprint: ProjectBlueprint) = projectBlueprint.allModuleBlueprints.joinToString("\n", "digraph ${projectBlueprint.projectName} {\n", "\n}") {
+        getDependencyForModuleAsString(it.name, projectBlueprint.allDependencies[it.name]?: listOf())
+    }
+
+    private fun getDependencyForModuleAsString(name: String, dependencies: List<ModuleDependency>): String {
+        var list = ""
+        if (dependencies.isNotEmpty()) {
+            list = " -> ${dependencies.map { it.name }.sorted().joinToString()}"
+        }
+
+        return "  $name$list;"
+    }
+
+    class DependencyImageGenerator(private val imageWriter: ImageWriter) {
+        companion object {
+            const val CELL_SIZE = 3
+            const val LINE_WIDTH = 1
+            const val HEADER_SIZE = CELL_SIZE + 2 * LINE_WIDTH
+            const val GRID_SIZE = CELL_SIZE + LINE_WIDTH
+            val BACKGROUND_COLOR = Color(0x000000)
+            val GRID_COLOR = Color(0x8F8F8F)
+            val APP_COLOR = Color(0x5050FF)
+            val ANDROID_COLOR = Color(0xA4C639)
+            val JAVA_COLOR = Color(0xF7DB64)
+            val ERROR_COLOR = Color(0xDF0000)
+            // Colors match https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph
+            val dependencyMethodToColor = mapOf(
```

Comment:
There might be a problem with flavours, but we'll tackle it later.

Reply:
Yes, different flavors can have different dependencies, we can either have all dependencies for all flavors in this image or generate one per variant. I think it would be better to have a single image because this image purpose is to have a summary of the dependencies.